components/AppSearch.vue

```diff
@@ -25,9 +25,30 @@ const { activate, deactivate } = useFocusTrap(searchContentRef)
 
 const { meta_K, Escape } = useMagicKeys()
 
-const { data: files } = await useLazyAsyncData<BlogSearchResult[]>(
-    'search-api',
-    () => $fetch('/api/search', { parseResponse: JSON.parse })
+const SEARCH_DATA_KEY = 'search-api';
+
+// Cache results across pages
+// Note: does not cache across client instances, use Cache-Control headers or local storage instead
+const cachedResults = useNuxtData<BlogSearchResult[]>(SEARCH_DATA_KEY)
+
+function showSearch() {
+    show.value = true;
+    if(!!!cachedResults.data.value || (cachedResults.data.value.length === 0)){
```

Comment:
I'm not sure to understand this part.

The search route is prerendered at build time. It should create a huge json file, included in every page that use this component

Is it useful to cache the result of this API ? (it's already done once, no) ?

Reply:
While the search route is prerendered at build time, it is not included in the page payload due to SSR process not fetching the information at build time on a page. 

The size of the page is kept minimal as the results are not required for page rendering until the user activates search, reducing the initial page loading time.

This is due to setting immediate to false and the execute function is only called by an action not performed in the server render.

The client instance needs to fetch the  data as it is not included in the payload. 

Instead of fetching the data each time when a user navigates between pages (CSR mode) and uses search, the useNuxtData caches the result in the client instance. The logic controls when to execute the fetch based on the initial cache contents being empty.

The result is not cached between client instances. See the search api code comments on how it could be cached across instances.

Reply:
ok, I understand
Thanks for your explanation !